### PR TITLE
LVM-425 updates to Coordinator (MVC and view tests)

### PR DESCRIPTION
**Pull request by jlally21**
https://cs5500-jira.ccs.neu.edu/browse/LVM-425

After updating the Coordinator schema, I needed to make updates/changes to the model, controller, views and all view tests

### CODE REVIEW

File: app/views/coordinators/index.html.erb [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
-      <td><%= link_to 'Show', coordinator_path(coordinator) %></td>
-      <td><%= link_to 'Edit', edit_coordinator_path(coordinator) %></td>
-      <td><%= link_to 'Destroy', coordinator_path(coordinator),
-        method: :delete,
-        data: { confirm: 'Are you sure?' } %></td>
+      <th>Name</th>
+      <th>Date of Birth</th>
+      <th>Email</th>
+      <th>Phone Number</th>
+      <th colspan="3"></th>
```
Comment (zack) [inline app/views/coordinators/index.html.erb]: :o

Comment (jlally21) [reply app/views/coordinators/index.html.erb]: @zack  ?

Comment (zack) [reply app/views/coordinators/index.html.erb]: What's the purpose of this?

File: app/views/coordinators/index.html.erb [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
+  <thead>
     <tr>
-      <td><%= coordinator.name %></td>
-      <td><%= link_to 'Show', coordinator_path(coordinator) %></td>
-      <td><%= link_to 'Edit', edit_coordinator_path(coordinator) %></td>
-      <td><%= link_to 'Destroy', coordinator_path(coordinator),
-        method: :delete,
-        data: { confirm: 'Are you sure?' } %></td>
+      <th>Name</th>
+      <th>Date of Birth</th>
```
Comment (zack) [inline app/views/coordinators/index.html.erb]: Do we really need dob on the index page? What use does it serve?

File: app/views/coordinators/edit.html.erb [hunk 3]
```diff
@@ -1,35 +1,6 @@
...
-  </p>
-
-  <p>
-    <%= f.submit %>
-  </p>
-
-<% end %>
-
-<%= link_to 'Back', coordinators_path %>
+<%= link_to 'Show', @coordinator %> |
```
Comment (zack) [inline app/views/coordinators/edit.html.erb]: I think we should call this "Cancel"

**Comment by zack** [comment]
this is an awkward error message (since it's not actually blank), but it might be a ticket for another day.

screen shot 2017-03-17 at 5 18 01 pm

**Comment by zack** [comment]
If we do keep DOB we probably want to space these columns a bit better.

screen shot 2017-03-17 at 5 19 14 pm

File: spec/views/coordinators/index_spec.rb [hunk 4]
```diff
@@ -1,62 +1,16 @@
...
-      end
-    end
+  it 'renders a list of coordinators' do
+    render
+    expect(rendered).to match(@coordinator.name)
+    expect(rendered).to match(@coordinator.date_of_birth.strftime('%Y-%m-%d'))
+    expect(rendered).to match(/\(999\) 999-9999/)
+    expect(rendered).to match(@coordinator.email)
   end
 end
```
Comment (zack) [inline spec/views/coordinators/index_spec.rb]: I would like to see a test that ensures this page shows multiple coordinators, but it's not a huge deal.

File: spec/views/student/show_spec.rb [hunk 5]
```diff
@@ -1,42 +1,41 @@
 require 'rails_helper'
 
-RSpec.describe 'coordinators/show.html.erb', type: :view do
+RSpec.describe 'students/show.html.erb', type: :view do
```
Comment (zack) [inline spec/views/student/show_spec.rb:3]: Did you pend these tests until we resolve how we related tutors and coordinators?

**Comment by zack** [comment]
I had a few comments (sorry) but this looks awesome! And I really dig the new errors!

**Comment by zack** [comment]
The only other thing is that the phone validations are pretty specific and so we should assume that nobody can actually use this form (as a user, not as a developer) until we implement masking so that they know what format we want. Which is fine; it's not in production, but I think it would be prudent to create that and assign that ticket before we merge this so we don't forget.

File: app/views/coordinators/edit.html.erb (hunk 3 above)
Comment (jlally21) [reply app/views/coordinators/edit.html.erb]: agreed

File: app/views/coordinators/index.html.erb (hunk 2 above)
Comment (jlally21) [reply app/views/coordinators/index.html.erb]: no we don't.  I'll remove

**Comment by jlally21** [comment]
created a ticket for phone number masking

File: spec/views/coordinators/index_spec.rb (hunk 4 above)
Comment (jlally21) [reply spec/views/coordinators/index_spec.rb]: added test for multiple coordinators

**Pull request merged**